pkg/connector/datasets.go

```diff
@@ -145,57 +162,59 @@ func (o *datasetBuilder) Grants(ctx context.Context, resource *v2.Resource, pTok
 	policy, err := o.projectsClient.GetIamPolicy(ctx, &iampb.GetIamPolicyRequest{
 		Resource: fmt.Sprintf("projects/%s", o.bigQueryClient.Project()),
 	})
-	if err != nil {
-		return nil, "", nil, wrapError(err, "Unable to fetch IAM policy for project")
+	if !isPermissionDenied(ctx, err) {
+		return nil, "", nil, wrapError(err, "listing grants for datasets failed")
 	}
 
 	var grants []*v2.Grant
-	for _, access := range dataset.Access {
-		if access.Role == bigquery.OwnerRole || access.EntityType == bigquery.UserEmailEntity {
-			g, err := o.GetUserOwnerGrants(resource, access)
-			if err != nil {
-				return nil, "", nil, err
+	if policy != nil {
```

Comment:
Early return if policy == nil. it makes the code easier to read.

Reply:
Fixed